File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if(l1.next==null & l1.val==0){
            return l2;
        }
        if(l2.next==null & l2.val==0){
            return l1;
        }
        ListNode ans = new ListNode();
        ListNode c1 = l1, c2 = l2, c3 = ans;
        int carry = 0;
        while(c1!=null || c2!=null){
            if(c1!=null & c2!=null){
                int num = c1.val+c2.val+carry;
                c3.next = new ListNode(num%10);
                carry = num/10;
                c1 = c1.next;
                c2 = c2.next;
            }
            else if(c1==null){
                int num = c2.val+carry;
                c3.next = new ListNode(num%10);
                carry = num/10;
                c2 = c2.next;
            }
            else if(c2==null){
                int num = c1.val+carry;
                c3.next = new ListNode(num%10);
                carry = num/10;
                c1 = c1.next;
            }
            c3 = c3.next;
        }
        if(carry>0){
            c3.next = new ListNode(carry);
        }
        return ans.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        if l1.next==None and l1.val==0:
            return l2
        if l2.next==None and l2.val==0:
            return l1
        c1, c2 = l1, l2
        carry = 0
        ans = ListNode()
        c3 = ans
        while(c1!=None or c2!=None):
            if(c1!=None and c2!=None):
                num = c1.val+c2.val+carry
                c3.next = ListNode(num%10)
                carry = num//10
                c1 = c1.next
                c2 = c2.next
            elif(c1==None):
                num = c2.val+carry
                c3.next = ListNode(num%10)
                carry = num//10
                c2 = c2.next
            elif(c2==None):
                num = c1.val+carry
                c3.next = ListNode(num%10)
                carry = num//10
                c1 = c1.next
            c3 = c3.next
        if(carry>0):
            c3.next = ListNode(carry)
            c3 = c3.next
        return ans.next
